packages/core/src/Table/Table.js

```diff
@@ -1,56 +1,59 @@
 import propTypes from '@dhis2/prop-types'
-import React from 'react'
-import css from 'styled-jsx/css'
-import { Provider } from './TableContext.js'
-
-const tableStyles = css`
-    table {
-        border: 1px solid #e8edf2;
-        background-color: #ffffff;
-        min-width: 100%;
-        text-align: left;
-        border-collapse: collapse;
-        vertical-align: top;
-    }
-`
+import { colors } from '@dhis2/ui-constants'
+import React, { forwardRef } from 'react'
 
 /**
  * @module
  * @param {Table.PropTypes} props
  * @returns {React.Component}
- * @example import { Table } from '@dhis2/ui-core'
- * @see Live demo: {@link /demo/?path=/story/table--static-layout|Storybook}
+ * @example import { Table } from '@dhis2/ui'
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/organisms/data-table.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/table--default|Storybook}
  */
-export const Table = ({
-    children,
-    className,
-    dataTest,
-    suppressZebraStriping,
-}) => (
-    <Provider value={{ suppressZebraStriping }}>
-        <table className={className} data-test={dataTest}>
-            {children}
-
-            <style jsx>{tableStyles}</style>
-        </table>
-    </Provider>
+export const Table = forwardRef(
+    ({ children, className, dataTest, layout, width }, ref) => {
+        return (
+            <table className={className} data-test={dataTest} ref={ref}>
+                {children}
+                <style jsx>{`
+                    table {
+                        table-layout: ${layout};
+                        border-collapse: separate;
+                        border-spacing: 0;
+                        width: ${width};
+                        box-sizing: border-box;
+                        border: 1px solid ${colors.grey300};
+                    }
+                `}</style>
+            </table>
+        )
+    }
 )
 
+Table.displayName = 'Table'
+
 Table.defaultProps = {
     dataTest: 'dhis2-uicore-table',
+    width: '100%',
+    layout: 'auto',
 }
 
 /**
  * @typedef {Object} PropTypes
  * @static
  * @prop {TableHead|TableBody|TableFoot|Array.<TableHead|TableBody|TableFoot>} [children]
  * @prop {string} [className]
- * @prop {string} [dataTest]
- * @prop {bool} [suppressZebraStriping]
+ * @prop {string} [dataTest=dhis2-uicore-table]
+ * @prop {(auto|fixed)} [layout=auto] Sets the `table-layout` property. Switching to `fixed` can prevent
+ *  style issues when dealing with a table with multiple frozen columns or when dealing with filter
+ *  elements in the table headers.
+ * @prop {string} [width=100%] Sets the `width` property. Providing an explicit width can prevent style
+ *  issues when dealing with horizontally scrolling tables with a fixed layout.
```

Comment:
Automatic table-layout is quite clever in terms of determining column width etc. However, when you are doing things like trying to apply a specific column width, or conditionally rendering elements in a cell which impacts column width, then automatic table-layout can start working against you....

To combat this, I added the `layout` prop which can be used to switch the table-layout to fixed. But this table-layout comes with a new set of issues of its own, specifically it will allow columns to shrink beyond the width required by its content. This is a problem when you implement a horizontally scrolling table and this problem can be avoided by explicitly setting a table-width (which isn't such a crazy thing to do anyway when you want to implements horizontal scrolling).

So yeah, there you have it: two pretty basic props which simply control some CSS. But the bigger picture is much more complex. I'm not sure how we would want to document this.

Also, another note about `table-layout: fixed`: I am considering implementing this as a `fixedLayout` prop of type `boolean` instead. What would have your preference?

Reply:
> What would have your preference?

@ismay, you, me and some others talked about boolean props vs string values quite a while ago and, **if I remember correctly**, we agreed that boolean props are nicer for boolean values. I'm not entirely sure though, but I think we should try to keep this consistent across our UI library

Reply:
Yeah we've talked about this general topic quite often. We have chosen to implement some properties as boolean even though they aren't really booleans, such as `error`, `warning` and `valid`. We chose to stick with them and even have the custom prop-type called `mutuallyExclusive`.

I had a quick search in the codebase to see where we are using `propTypes.oneOf()` to scan comparable scenarios but couldn't really find one that is super close to this.

In general I think boolean props work for discrete choices, which I think is a slightly broader definition than just saying boolean props are suitable for boolean values. Suppose there were only two values for the CSS `table-layout` property, one called `auto` and the other called `fixed`, where `auto` was the default: in this case I would say this is a 100% fit with a boolean prop called `fixed` or `fixedLayout`.

This nearly true for `table-layout` but not quite because there is also `inherit` and `initial`, which aren't values you'd use very often.

To me both of the following options would be valid solutions:
1. Keep `layout` as a string prop but allow all valid CSS values `auto` `fixed` `initial` and `inherit`.
2. Replace with a boolean prop called `fixedLayout` / `layoutFixed`.

I guess the underlying choice is whether or not we expect we ever need to set a table-layout to anything else but `auto` or `fixed`. 

After re-reading my own reasoning above I'm inclined to go with option 1.

Reply:
> After re-reading my own reasoning above I'm inclined to go with option 1.

:+1: